app/controllers/concerns/waste_carriers_engine/can_redirect_form_to_correct_path.rb

```diff
@@ -13,6 +13,9 @@ def redirect_to_correct_form
       # new_state_name_path/:token or start_state_name_path?token=:token
       def form_path
         send("new_#{@transient_registration.workflow_state}_path".to_sym, token: @transient_registration.token)
+      rescue NoMethodError
+        # if unable to find the route in the engine, let the app try to resolve it
+        main_app.send("new_#{@transient_registration.workflow_state}_path".to_sym, token: @transient_registration.token)
       end
```

Comment:
Is this expected behaviour or is this rescue added just as a safety measure? 

- If we're expecting this rescue to be hit in normal circumstances, probably better to refactor it with if-else

- If we're not expecting the rescue to be hit in normal circumstances, then some error logging here would be handy.